sql/core/src/test/scala/org/apache/spark/sql/JoinSuite.scala

```diff
@@ -1086,4 +1087,21 @@ class JoinSuite extends QueryTest with SharedSparkSession with AdaptiveSparkPlan
       assert(df2.join(df1, "id").collect().isEmpty)
     }
   }
+
+  test("SPARK-32330: Preserve shuffled hash join build side partitioning") {
+    withSQLConf(
+        SQLConf.AUTO_BROADCASTJOIN_THRESHOLD.key -> "50",
```

Comment:
nit: set it to "-1" to make the intention (turning off broadcast join) clear?

Reply:
@imback82 [query planner depends on this config to be carefully tuned here to trigger shuffled hash join](https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/joins.scala#L353-L355).

Reply:
Ah OK. Thanks!